app/src/test/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpassTest.kt

```diff
@@ -1011,11 +1040,85 @@ class HikeRoutesRepositoryOverpassTest {
       callbackCapture.firstValue.onResponse(mockCall, simpleResponse)
     }
 
+    hikingRouteProviderRepositoryOverpass.getRouteById(
+        simpleRoutes.first().id,
+        { assertEquals(simpleRoutes.first(), it) },
+        { fail("onFailure shouldn't have been called") })
+  }
+
+  @Test
+  fun getRoutesByIds_hasEmptyListOnEmptyResponse() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, emptyResponse)
+    }
+
+    hikingRouteProviderRepositoryOverpass.getRoutesByIds(
+        listOf(simpleRoutes.first().id),
+        { assertEquals(emptyList<HikeRoute>(), it) },
+        { fail("onFailure shouldn't have been called") })
+  }
+
+  @Test
+  fun getRoutesByIds_failsOnFailedResponse() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, failedResponse)
+    }
+
+    var failCalled = false
+
+    hikingRouteProviderRepositoryOverpass.getRoutesByIds(
+        listOf(simpleRoutes.first().id),
+        { fail("onSuccess shouldn't have been called") },
+        { failCalled = true })
+
+    assert(failCalled)
+  }
+
+  @Test
+  fun getRoutesByIds_succeedsOnSingleRoute() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, simpleResponse)
+    }
+
     var successCalled = false
 
-    hikingRouteProviderRepositoryOverpass.getRouteById(
-        "124582", { successCalled = true }, { fail("onFailure shouldn't have been called") })
+    hikingRouteProviderRepositoryOverpass.getRoutesByIds(
+        listOf(simpleRoutes.first().id),
+        { successCalled = true },
+        { fail("onFailure shouldn't have been called") })
 
     assert(successCalled)
   }
+
+  @Test
+  fun getRoutesByIds_succeedsOnMultipleRoutes() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, doubleResponse)
+    }
+
+    hikingRouteProviderRepositoryOverpass.getRoutesByIds(
+        listOf(doubleRoutes[0].id, doubleRoutes[1].id),
+        { assertEquals(doubleRoutes, it) },
```

Comment:
Nitpick: you did not use `successCalled` here either, the function might call neither `onSuccess` nor `onFailure` and the test would still pass.

Reply:
Thanks for it!